docs/developer/release-process.md

```diff
@@ -2,67 +2,154 @@
+
+> **Note**: this image is not being built automatically. Consequently, a [manual build process](./end-to-end-tests.md#building-the-"kubeapps/integration-tests"-image) _must_ be triggered if you happen to upgrade the integration image.
+
+### 0.3 - Development chart
+
+Even though the official [Bitnami chart](https://github.com/bitnami/charts/tree/master/bitnami/kubeapps) is automatically able to retrieve the latest dependency versions, we still need to sync the versions declared in our own development chart.
+
+#### 0.3.1 - Chart images
+
+Currenty, the [values.yaml](../../chart/kubeapps/values.yaml) uses the following container images:
+
+- [bitnami/nginx](https://hub.docker.com/r/bitnami/nginx/tags)
+- [bitnami/kubectl](https://hub.docker.com/r/bitnami/kubectl/tags)
+- [bitnami/oauth2-proxy](https://hub.docker.com/r/bitnami/oauth2-proxy/tags)
+
+> As part of this release process, these image tags _must_ be updated to the latest minor version. In case of a major version, the change _should_ be tracked in a separate PR.
+
+#### 0.3.2 - Chart dependencies
+
+The chart [requirements.yaml](../../chart/kubeapps/requirements.yaml) _must_ be checked to ensure the version includes the latest dependent charts.
+
+- Check if the latest versions are already included by running:
 
 ```bash
 helm dependency list ./chart/kubeapps
 ```
 
-to see if the latest versions are included, and
+- If they are not, run this other command to update the `requirements.lock` file:
 
 ```bash
 helm dependency update ./chart/kubeapps
 ```
 
-to update the requirements.lock file.
+> As part of this release process, the chart dependencies _must_ be updated to the latest versions. In case of a major version, the change _should_ be tracked in a separate PR.
+
+### 0.4 - Upgrading the code dependencies
+
+Currently, we have three types of dependencies: the [dashboard dependencies](../../dashboard/package.json), the [golang dependencies](../../go.mod), and the [rust dependencies](../../cmd/pinniped-proxy/Cargo.toml). They _must_ be upgraded to the latest minor/patch version to get the latest bug and security fixes.
 
-Next, run
+- Upgrade the [dashboard dependencies](../../dashboard/package.json) by running:
 
 ```bash
+cd dashboard
 yarn upgrade
 ```
 
-in the dashboard directory will update the frontend packages to the latest compatible versions.
+- Check the outdated [golang dependencies](../../go.mod) by running:
 
-Finally, look at the [pull requests](https://github.com/kubeapps/kubeapps/pulls) and ensure there is no PR open by Snyk fixing a security issue.  If so, discuss it with a peer and come to a decision on it, trying not to release with a high/medium severity issue.
+```bash
+go mod tidy
+go list -u -f '{{if (and (not (or .Main .Indirect)) .Update)}}{{.Path}}: {{.Version}} -> {{.Update.Version}}{{end}}' -m all 2> /dev/null
```

Comment:
Woah - did you create that, or is that a recommendation from somewhere (if so, where, and can you link to the source?)